File: tests/fixtures/sample-python-project-negative/shared/utils/formatters.py
from datetime import datetime


# VIOLATION: style/deterministic/docstring-completeness
def format_user(user: dict) -> dict:
    return {
        "id": user["id"],
        "name": user["name"],
        "email": user["email"],
        "displayName": f"{user['name']} <{user['email']}>",
        "createdAt": datetime.fromisoformat(str(user["createdAt"])).isoformat(),
    }


# VIOLATION: style/deterministic/docstring-completeness
# VIOLATION: code-quality/deterministic/missing-maxsplit-arg
def format_date(date: datetime) -> str:
    return date.isoformat().split("T")[0]

File: tests/fixtures/sample-js-project-positive-old/shared/utils/src/formatters.ts
interface RawUser {
  id: string;
  name: string;
  email: string;
  createdAt: string | Date;
}

export function formatUser(user: RawUser): { id: string; name: string; email: string; displayName: string; createdAt: string } {
  return {
    id: user.id,
    name: user.name,
    email: user.email,
    displayName: `${user.name} <${user.email}>`,
    createdAt: new Date(user.createdAt).toISOString(),
  };
}

export function formatDate(date: Date): string {
  const parts = date.toISOString().split('T');
  return parts[0] ?? '';
}
